### ISSUE TITLE
Flutter 3.7.11 Xcode 14.3 Failed to build iOS app

### ISSUE THREAD

**Issue by YaoShuwei** [flutter/flutter#124835]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. create a scheme named "dev"
2. run "flutter run --flavor=dev"

### Expected results

Successful run on the iOS Simulator

### Actual results

Failed to build iOS app

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
Launching lib/main.dart on iPhone 14 Pro in debug mode...
Running Xcode build...
Xcode build done.                                            2.7s
Failed to build iOS app
Error output from Xcode build:
↳
    2023-04-14 10:34:07.928 xcodebuild[28647:219942] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)
    xcodebuild: error: Unable to find a destination matching the provided destination specifier:
    		{ id:C311C176-8DD0-47E6-97AF-24D51BEEC8BB }

    	Available destinations for the "dev" scheme:
    		{ platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device }

Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/xr/6lbh_mk957ngr87s6bw78_dw0000gn/T/flutter_tools.n4NdNq/flutter_ios_build_temp_dir9043NQ/temporary_xcresult_bundle


Could not build the application for the simulator.
Error launching application on iPhone 14 Pro.
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.7.11, on macOS 13.2.1 22D68 darwin-x64, locale zh-Hans-CN)
    • Flutter version 3.7.11 on channel stable at /Users/apaipai/fvm/versions/3.7.11
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f72efea43c (2 days ago), 2023-04-11 11:57:21 -0700
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/apaipai/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • ANDROID_SDK_ROOT = /Users/apaipai/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 73.0.4
    • Dart plugin version 231.8109.91

[✓] Connected device (4 available)
    • iPhone 14 Pro (mobile) • C311C176-8DD0-47E6-97AF-24D51BEEC8BB     • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)        • macos                                    • darwin-x64     • macOS 13.2.1 22D68 darwin-x64
    • Chrome (web)           • chrome                                   • web-javascript • Google Chrome 110.0.5481.100

[✓] HTTP Host Availability
    • All required HTTP hosts are available

```

</details>

**Comment by JherysVargas** [comment]
I have the same problem, any solution?

**Comment by atleugim** [comment]
Samee here

**Comment by huycozy** [comment]
@YaoShuwei It seems the issue is caused by deployment target. Please check [this](https://github.com/flutter/flutter/issues/124340#issuecomment-1500587231) if it can help. 

@JherysVargas @atleugim Are you also seeing this issue on the latest stable 3.7.11?

**Comment by YaoShuwei** [comment]
> @YaoShuwei It seems the issue is caused by deployment target. Please check [this](https://github.com/flutter/flutter/issues/124340#issuecomment-1500587231) if it can help.
> 
> @JherysVargas @atleugim Are you also seeing this issue on the latest stable 3.7.11?
My deployment is already on ios 11.
There is no error when I launch app without flavor.

**Comment by huycozy** [comment]
@YaoShuwei I also try with [flavor sample](https://github.com/flutter/flutter/tree/master/dev/integration_tests/flavors) on XCode 14.3 but still can not reproduce this issue. Please also try running it to see if it works or not.

Also, can you share your Podfile content?

**Comment by YaoShuwei** [comment]
<details><summary>Podfile Content</summary>

```console
# Uncomment this line to define a global platform for your project
platform :ios, '11.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

# post_install do |installer|
#   installer.pods_project.targets.each do |target|
#     flutter_additional_ios_build_settings(target)
#   end
# end
post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
    target.build_configurations.each do |config|
      config.build_settings['ENABLE_BITCODE'] = 'NO'
      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '11.0'
      config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= [
          '$(inherited)',

          ## dart: PermissionGroup.calendar
          # 'PERMISSION_EVENTS=1',

          ## dart: PermissionGroup.reminders
          # 'PERMISSION_REMINDERS=1',

          ## dart: PermissionGroup.contacts
          'PERMISSION_CONTACTS=1',

          ## dart: PermissionGroup.camera
          'PERMISSION_CAMERA=1',

          ## dart: PermissionGroup.microphone
          # 'PERMISSION_MICROPHONE=1',

          ## dart: PermissionGroup.speech
          # 'PERMISSION_SPEECH_RECOGNIZER=1',

          ## dart: PermissionGroup.photos
          'PERMISSION_PHOTOS=1',

          ## dart: [PermissionGroup.location, PermissionGroup.locationAlways, PermissionGroup.locationWhenInUse]
          'PERMISSION_LOCATION=1',

          ## dart: PermissionGroup.notification
          # 'PERMISSION_NOTIFICATIONS=1',

          ## dart: PermissionGroup.mediaLibrary
          # 'PERMISSION_MEDIA_LIBRARY=1',

          ## dart: PermissionGroup.sensors
          # 'PERMISSION_SENSORS=1',

          ## dart: PermissionGroup.bluetooth
          # 'PERMISSION_BLUETOOTH=1',

          ## dart: PermissionGroup.appTrackingTransparency
          # 'PERMISSION_APP_TRACKING_TRANSPARENCY=1',

          ## dart: PermissionGroup.criticalAlerts
          # 'PERMISSION_CRITICAL_ALERTS=1'
        ]
    end
  end
end

```

</details>

**Comment by huycozy** [comment]
I don't see your flavor configuration on your Podfile. Please see [Creating flavors for Flutter](https://docs.flutter.dev/deployment/flavors) and make sure you have completed all steps there. 

You may also try the flavor sample project in my previous comment as well.

**Comment by Rollfreeze** [comment]
Same here guys, I have the same problem now

**Comment by JherysVargas** [comment]
@huycozy I confirm that I have the latest version and the error still occurs

<img width="608" alt="Captura de pantalla 2023-04-14 a la(s) 9 12 33 a m" src="https://user-images.githubusercontent.com/39938693/232070075-f4adc2e7-ea00-4012-9a4e-b16000519195.png">

**Comment by stayf** [comment]
This is a branch from release 3.7.10 https://github.com/flutter/engine/tree/flutter-3.7-candidate.1
This is a branch from release 3.7.11 https://github.com/flutter/engine/tree/stable And this is not an engine build at all.

**Comment by JherysVargas** [comment]
@stayf I don't understand your answer.

Has anyone managed to solve the problem?

**Comment by huycozy** [comment]
@atleugim @JherysVargas Thanks for your response. But are you only have this issue with flavor? 

I tried [flavor sample](https://github.com/flutter/flutter/tree/master/dev/integration_tests/flavors) on XCode 14.3 but can not reproduce this issue. Can you please also try running that sample to see if the issue can be reproducible?

**Comment by figureai** [comment]
<img width="679" alt="image" src="https://user-images.githubusercontent.com/45620405/233256493-c9a74073-61e9-4e44-985f-d24726d52d4c.png">
I have the same problem.

**Comment by huycozy** [comment]
@figureai You are using an old Flutter version. Please retry on the latest stable Flutter version (3.7.12)

**Comment by figureai** [comment]
<img width="647" alt="image" src="https://user-images.githubusercontent.com/45620405/233275980-aa98958b-62f3-48b3-803d-0a44c7bc514b.png">
I upgrade to the latest stable Flutter version(3.7.12), but it is still have the same error

**Comment by VanTy207** [comment]
I have the same problem.

**Comment by DJuan33** [comment]
Same problem. Flutter 3.7.12:
![imagen](https://user-images.githubusercontent.com/45220181/235219790-c4f91074-59b7-47a6-8fda-e3c297566bdd.png)

**Comment by shenhuaxiyuan** [comment]
same problem too !

**Comment by rivella50** [comment]
I have the same problem. I have 2 XCode versions installed (using Xcodes app for switching between 14.2 and 14.3).
As soon as i go back to version 14.2 the connected iPhone (with OS version 16.4.1) is listed when using `fvm flutter devices`, and vanishes when using version 14.3.
Of course deploying my Flutter app also doesn't work (simulator and physical device) with the same error @DJuan33 mentioned above.
What is interesting: When opening the project in XCode 14.3 the iPhone is listed and i even can build and deploy my app without any problems.

When connecting an older iPhone (with OS version 15.7.5) that one is listed using XCode 14.3.

I tried using Flutter versions 3.7.11 and 3.7.12.
Cocoapods version is still version 1.11.3.

**Comment by huycozy** [comment]
Are you only seeing this issue when using flavor as mentioned above?

I tried reproducing this issue with [flavor example](https://github.com/flutter/flutter/tree/master/dev/integration_tests/flavors) on Xcode 14.3, CocoaPods 1.11.3 and latest Flutter stable channel `3.7.12` but can't see the issue. 

Verifying this on `iPhone 7, iOS 15.7.2 (physical device)` and `iPhone 14 Pro, iOS 16.4 (emulator)`.

<details>
<summary>Demo</summary>

<img width="400" src="https://user-images.githubusercontent.com/104349824/236177888-1f972678-e56e-4d28-babb-85c396229ddd.png">

</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.7.12, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.7.12 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4d9e56e694 (2 days ago), 2023-04-17 21:47:46 -0400
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.77.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.62.0

[✓] Connected device (3 available)
    • SM T225 (mobile) • R9JT3004VRJ • android-arm64  • Android 13 (API 33)
    • macOS (desktop)  • macos       • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)     • chrome      • web-javascript • Google Chrome 112.0.5615.137

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.11.0-1.0.pre.43, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.11.0-1.0.pre.43 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 15386d9d17 (21 minutes ago), 2023-05-03 21:35:35 -0400
    • Engine revision ee65be7d8f
    • Dart version 3.1.0 (build 3.1.0-70.0.dev)
    • DevTools version 2.23.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.77.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.62.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 112.0.5615.137

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

---

But I just saw there is a possibly related issue (https://github.com/flutter/flutter/issues/124408) and it has been fixed recently. Can you please retry on the latest master channel to see if the issue is fixed or not?

```
flutter channel master
flutter upgrade
```

**Comment by rivella50** [comment]
@huycozy I would say on my side it's independent of flavors.
I've just tried with the most recent version fvm offers (3.10.0-1.5.pre) and the iPhone with version 16.4.1 is still not discovered when XCode 14.3 is active.

It would be interesting to know what components are involved when `flutter devices` is called.

**Comment by rivella50** [comment]
When calling `flutter devices -v` i see that this call happens:
`/usr/bin/arch -arm64e xcrun xcdevice list --timeout 2`

which results in this error (XCode 14.3 active) besides the list of found iOS simulators:
`2023-05-04 13:04:44.942 xcdevice[45023:439498] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)`

This is the error from above and actually has nothing to do with Flutter i would say (or only indirectly).
When XCode 14.2 is active this error doesn't happen.

**Comment by huycozy** [comment]
>I've just tried with the most recent version fvm offers (3.10.0-1.5.pre) 

@rivella50 The above version is on beta channel, please also check on the latest master as well. 

Also, could you check if the issue happens on a newly created Flutter project? If it does, please provide the complete output of `flutter run -v` so that we may verify the issue. Thanks!

**Comment by rivella50** [comment]
@huycozy I just tried with version Flutter 3.11.0-1.0.pre.63: no change.
As mentioned in my last post i don't think that the problem has to do with Flutter.
What happens when you call `/usr/bin/arch -arm64e xcrun xcdevice list --timeout 2` with XCode 14.3 active and a physical iOS device with OS version 16.4.x ?

**Comment by rivella50** [comment]
To clarify: The error we all encounter here does not happen because the build fails but because the targeted device (simulator or physical with iOS 16.4.x) cannot be found (although it's connected).
I am capable of building an IPA file using XCode 14.3, but i'm not able to deploy a build.
Therefore it doesn't make sense to test with different Flutter projects.

When trying to call `flutter run ...` the mentioned `xcrun xcdevice` seems to be perfomed in the background in order to get the list of available devices and verify if the targeted device is found as well which then leads to the mentioned error above.

To me it seems that devices with iOS versions 16.4.x are not registered/recognized when using XCode 14.3, where devices with older iOS versions (e.g. 15.7.x) do show up and can be used without problems for a deployment.

**Comment by DavidIjsud** [comment]
here the same. i even created a new project and added the [camera](https://pub.dev/packages/camera) plugin as the unique dependence of the new project then i run it on ios emulator but it stop. 

<img width="1050" alt="Screenshot 2023-05-06 at 18 48 52" src="https://user-images.githubusercontent.com/35741206/236649360-6287a490-9367-49a3-a250-82be934bb14c.png">

<img width="1310" alt="Screenshot 2023-05-06 at 18 56 31" src="https://user-images.githubusercontent.com/35741206/236649526-b5a70ead-b8f7-49cc-8c12-122c59c2b020.png">

**Comment by rivella50** [comment]
Regarding the simulators: What works for me is edit the scheme(s) in XCode 14.3 and change the value from 'release' to 'debug' for the Build Configuration.
With that change the app runs fine again.
But this doesn't solve the problem of not showing the physical devices with OS 16.4.x loaded for calls like `flutter devices`.

**Comment by huycozy** [comment]
Thanks for your response. Since the OP doesn't seem to respond to this issue, I will label this for more insights and opinions. 

### Issue summary
-  The issue raised by OP is only seen when using flavor (see [this](https://github.com/flutter/flutter/issues/124835#issuecomment-1508166801))
- Some reports indicate the issue is still being seen on stable `3.7.12` ([comment](https://github.com/flutter/flutter/issues/124835#issuecomment-1515777769), [comment](https://github.com/flutter/flutter/issues/124835#issuecomment-1527904786), [comment](https://github.com/flutter/flutter/issues/124835#issuecomment-1537238135)) and also on master `3.11.0-1.0.pre.63` ([comment](https://github.com/flutter/flutter/issues/124835#issuecomment-1535725239))

/cc: @jmagman for thoughts (since I see the [comment](https://github.com/flutter/flutter/issues/123890#issuecomment-1502114849))

**Comment by rivella50** [comment]
I still try to narrow down the problem. Therefore i have two questions for you guys who encounter this error:
- do you have an Intel or an M1/M2 chip?
- do you have more than one Xcode version installed on your Mac and how did you install it/them (AppStore or Web download)?

Perhaps we can see a pattern here.

**Comment by YaoShuwei** [comment]
> I still try to narrow down the problem. Therefore i have two questions for you guys who encounter this error:
> 
> * do you have an Intel or an M1/M2 chip?
> * do you have more than one Xcode version installed on your Mac and how did you install it/them (AppStore or Web download)?
> 
> Perhaps we can see a pattern here.

chip： Intel
Xcode version: 14.3 (update from 14.2 ,AppStore)

When the issue happened, I download the 14.2 from web manually, it's no error there.

**Comment by rivella50** [comment]
@YaoShuwei Thanks for your response. Since the error happens with an M1 chip on my side (and i tried both Xcode 14.3 installed via AppStore and web download) i'd say there is no pattern.
Since we all encounter this error
`xcdevice[45023:439498] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)`
i wonder if there is a chance that we can modify this disabled state to an enabled one somehow, but i don't find any information about it.

**Comment by christopherfujino** [comment]
Seeing this also on CI https://ci.chromium.org/p/flutter/builders/try/Mac%20build_tests_2_4/43088?

**Comment by vashworth** [comment]
I'm unable to recreate this issue.

For those who are having this issue, can you please provide the following:
1. Please provide output from `flutter doctor -v`
2. What flutter command are you running that is causing the issue?
3. Please provide the error message.
4. What device are you using - Simulator or physical device? What iOS version?
5. Does your project use plugins? If so, what plugins?
6. Please go into Xcode > Runner > Project > Build Settings > Supported Platforms and check what Support Platforms are set (screenshot would be appreciated).

It would be helpful if someone who can recreate runs their command with `-v` and uploads the results as a txt file

**Comment by rivella50** [comment]
@vashworth Please read my [comment](https://github.com/flutter/flutter/issues/124835#issuecomment-1536606187) from above.
I still haven't received any feedback from Apple support.
From my latest observations i would say there is a problem in Command Line Tools in combination with Xcode 14.3.
All commentators above tried to run their Flutter applications from outside Xcode (command line or in another IDE) which means that at some point a determination of the available devices (simulator and physical) has to be done by using a command like `xcrun xcdevice list`. In my case the physical iPhone with iOS 16.4.1 isn't listed in the result which causes the error we all encounter.

**Comment by rivella50** [comment]
Btw. i just tried with the new Xcode 15 Beta, and it doesn't work with that version either.

**Comment by rivella50** [comment]
Has anybody of you been able to find a solution for this problem?
I got feedback from Apple and they seem not to be able to reproduce this issue on their side.
Since it also doesn't work with Xcode 15 we sooner or later will have a problem using still Xcode 14.2 i'd say.

**Comment by blueeyestw** [comment]
If you are using Xcode 13, this might help:

**Comment by rivella50** [comment]
@blueeyestw If the issue would be a build error this would be the solution, yes.
But we encounter a run error, where the destination device cannot be found.

**Comment by elliotnash** [comment]
> Since I'm unable to recreate this issue, if someone who is able to recreate could do `flutter run -v` and attach the results as a txt file here, that would help in debugging it.

I'm also getting this issue, here's the build log
[buildlog.txt](https://github.com/flutter/flutter/files/11780217/buildlog.txt)

**Comment by vashworth** [comment]
@elliotnash Thank you for the build log. I have an theory of what's wrong. Could you try opening your project in Xcode and in the Build Settings for the Runner target, change `ONLY_ACTIVE_ARCH` to `No`?

Example:
![Screenshot 2023-06-20 at 10 14 47 AM](https://github.com/flutter/flutter/assets/15619084/b012f77e-e3da-422b-8061-c7ce3f01a744)

Would you let me know if that fixes the issue for you?

**Comment by elliotnash** [comment]
Previously it would give that error linked above when building through flutter, so as a workaround I was running through xcode (which worked fine), but yesterday I managed to break the xcode builds too so I deleted and recreated the ios folder.

Just now when I tried to run it through flutter (with the recreated ios folder) it worked perfectly without making any changes -- even with these ONLY_ACTIVE_ARCH settings: ![Screenshot 2023-06-20 at 9 51 56 AM](https://github.com/flutter/flutter/assets/53949099/9151ecf2-3b16-46b6-8ae5-2c3625a0566c)

**Comment by prsh0525** [comment]
I don't know why this works, but I think I found how to solve it.

1. Open Xcode
2. Click "Runner" (root)
3. Click "Build Settings"
4. Change "Supported Platforms" from iphoneos to iOS

Make sure "Profile" and "Release" should be "iOS", not "iphoneos".

<img width="1041" alt="스크린샷 2023-06-21 오후 1 15 40" src="https://github.com/flutter/flutter/assets/10384956/ba129af1-1479-413d-b196-dbd377ae2b71">

**Comment by v11** [comment]
@prsh0525 This worked for me. I can not run from XCode but from VSCode with "flutter run" the project is starting in an IOS 16.4 simulator.

**Comment by ahmedalttai** [comment]
updating flutter to  3.10.5 fixed it for me.

**Comment by mustansirgodhrawala** [comment]
Still relevant, any idea when a fix will be available?

**Comment by SalarPro** [comment]
if you change the build target to **release**, reset it to **debug**, and it will work
open
ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme
change the buildConfiguration to "Debug" in side LaunchAction tag

before:
```
<LaunchAction
      buildConfiguration = "Release"
      ....
```
After:
```
<LaunchAction
      buildConfiguration = "Debug"
      .....
```

**Comment by arnavxjain** [comment]
Still experiencing the same issue, I'm currently on the latest stable version but the problem still persists. Any updates?

**Comment by sandrodz** [comment]
After Xcode update to 15.2 it started happening.

Resolved by:

<img width="594" alt="image" src="https://github.com/flutter/flutter/assets/8479569/d2a0c649-3301-40d2-b366-917c2539db84">

**Comment by jmagman** [comment]
@sandrodz Flutter doesn't run in Profile or Release mode on the simulator, so `SUPPORTED_PLATFORMS` being set to exclude the simulator was intentional.
https://github.com/flutter/flutter/pull/38905
https://github.com/flutter/flutter/issues/17735#issuecomment-523128904

**Comment by sandrodz** [comment]
@jmagman yes, but what do you suggest to do? without these, we get:

<img width="862" alt="image" src="https://github.com/flutter/flutter/assets/8479569/28b39dfa-bd57-4818-ad40-c4db625dcaaa">

**Comment by jmagman** [comment]
I suspect like you have your debug flavor set up incorrectly to use the Profile or Release version of Flutter. Make sure your `xcconfig` Configurations are set up correctly in Xcode (see note in [the docs](https://docs.flutter.dev/deployment/flavors#creating-flavors-in-ios)) 
<img width="753" alt="Screenshot 2024-01-18 at 6 30 24 PM" src="https://github.com/flutter/flutter/assets/682784/f56dcfd2-8963-45dd-856d-1d8235335040">


And make sure your Run action is set to a Debug build configuration:
<img width="923" alt="Screenshot 2024-01-18 at 6 30 36 PM" src="https://github.com/flutter/flutter/assets/682784/420114d0-75a4-438f-97a9-334467814810">

**Comment by sandrodz** [comment]
This did the trick, thanks.

<img width="1512" alt="image" src="https://github.com/flutter/flutter/assets/8479569/981e9c44-3589-450a-908d-7cd80612319a">

**Comment by duhaime** [comment]
I hit this when my ios version for xcode was out of date--I upgraded and unblocked myself with:

```
xcodebuild -downloadPlatform iOS
```

**Comment by blexo** [comment]
I just want you to know, if you still have this problem after having confirmed and/or tried every solution listed on this long, long thread:  you are not alone.
